lib/Dialect/FIRRTL/Transforms/BlackBoxMemory.cpp

```diff
@@ -226,12 +238,63 @@ static FModuleOp createWrapperModule(MemOp op,
          llvm::enumerate(memPortType.cast<BundleType>().getElements())) {
       auto fieldValue =
           builder.create<SubfieldOp>(op.getLoc(), memPort, field.index());
-      // Create the connection between module arguments and the external module,
-      // making sure that sinks are on the LHS
-      if (!field.value().isFlip)
-        builder.create<ConnectOp>(op.getLoc(), *extResultIt, fieldValue);
-      else
-        builder.create<ConnectOp>(op.getLoc(), fieldValue, *extResultIt);
+      auto fieldType = field.value().type.cast<FIRRTLType>();
+      bool isReadPort = field.value().isFlip;
+      // Flatten the vector, only if the elements of the vector are of simple
+      // ground type with known bitwidth.
+      if (fieldType.isa<FVectorType>() && flattenVector &&
+          fieldType.dyn_cast<FVectorType>()
```

Comment:
You can avoid these redundant type checks (including the `dyn_cast()` several lines lower) if you hoist the `dyn_cast()` above the `if` statement and check if the result is not `null`.